src/ray/core_worker/transport/raylet_transport.cc

```diff
@@ -29,4 +28,28 @@ Status CoreWorkerRayletTaskReceiver::GetTasks(std::vector<TaskSpec> *tasks) {
   return Status::OK();
 }
 
+CoreWorkerRayletTaskReceiver::CoreWorkerRayletTaskReceiver(
+    std::unique_ptr<RayletClient> &raylet_client, boost::asio::io_service &io_service,
+    rpc::GrpcServer &server)
+    : raylet_client_(raylet_client), task_service_(io_service, *this) {
+  server.RegisterService(task_service_);
+}
+
+void CoreWorkerRayletTaskReceiver::HandleAssignTask(
+    const rpc::AssignTaskRequest &request, rpc::AssignTaskReply *reply,
+    rpc::RequestDoneCallback done_callback) {
+  const std::string &task_message = request.task_spec();
+  const raylet::Task task(*flatbuffers::GetRoot<protocol::Task>(
+      reinterpret_cast<const uint8_t *>(task_message.data())));
+  const auto &spec = task.GetTaskSpecification();
+
+  auto status = task_handler_(spec);
+  done_callback(status);
+}
+
+Status CoreWorkerRayletTaskReceiver::SetTaskHandler(const TaskHandler &callback) {
+  task_handler_ = callback;
```

Comment:
Unless there is some reason we'd overwrite this, I think this should be set as part of the constructor to avoid calling an uninitialized `task_handler_`.

Reply:
I removed this function from this PR. Next pr would initialize the `task_handler_` in constructor.